src/repository/screens/commit.screen.js

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable react/no-array-index-key */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { View, ScrollView, Text, FlatList, StyleSheet } from 'react-native';
+import { Card } from 'react-native-elements';
+import {
+  ViewContainer,
+  DiffBlocks,
+  CodeLine,
+  LoadingContainer,
+} from 'components';
+import Parse from 'parse-diff';
+import { translate } from 'utils';
+import { colors, fonts, normalize } from 'config';
+import { getCommitDetails } from '../repository.action';
+
+const mapStateToProps = state => ({
+  language: state.auth.language,
+  commit: state.repository.commit,
+  diff: state.repository.diff,
+  isPendingCommit: state.repository.isPendingCommit,
+  isPendingDiff: state.repository.isPendingDiff,
+});
+
+const mapDispatchToProps = dispatch => ({
+  getCommitDetailsByDispatch: commit => dispatch(getCommitDetails(commit)),
+});
+
+const styles = StyleSheet.create({
+  fileChangeContainer: {
+    padding: 0,
+    marginTop: 12,
+    marginBottom: 12,
+  },
+  fileTitleContainer: {
+    flexDirection: 'row',
+    paddingVertical: 15,
+    backgroundColor: colors.greyVeryLight,
+  },
+  linesChanged: {
+    flex: 0.3,
+    paddingLeft: 10,
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  lineNumbersChanged: {
+    ...fonts.fontCode,
+    marginRight: 5,
+  },
+  fileTitle: {
+    flex: 1,
+    marginLeft: 10,
+  },
+  codeStyle: {
+    ...fonts.fontCode,
+    fontSize: normalize(10),
+  },
+  dividerStyle: {
+    marginBottom: 0,
+  },
+  noChangesMessage: {
+    ...fonts.fontPrimarySemiBold,
+    paddingVertical: 5,
+    paddingLeft: 10,
+  },
+  newIndicator: {
+    ...fonts.fontPrimarySemiBold,
+    color: colors.green,
+  },
+  deletedIndicator: {
+    ...fonts.fontPrimarySemiBold,
+    color: colors.red,
+  },
+  headerContainer: {
+    paddingTop: 15,
+    paddingHorizontal: 25,
+  },
+  header: {
+    flex: 1,
+    paddingTop: 10,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  headerItem: {
+    flex: 1,
+  },
+  headerText: {
+    ...fonts.fontPrimarySemiBold,
+    fontSize: normalize(14),
+  },
+});
+
+class Commit extends Component {
+  props: {
+    navigation: Object,
+    commit: Object,
+    diff: string,
+    isPendingCommit: boolean,
+    isPendingDiff: boolean,
+    getCommitDetailsByDispatch: Function,
+    language: string,
+  };
+
+  componentDidMount() {
+    const { navigation, getCommitDetailsByDispatch } = this.props;
+    const commit = navigation.state.params.commit;
+
+    getCommitDetailsByDispatch(commit);
```

Comment:
Just found a problem, I have:
```javacript
props: {
    navigation: Object,
    commit: Object,
    diff: string,
    isPendingCommit: boolean,
    isPendingDiff: boolean,
    getCommitDetailsByDispatch: Function,
    language: string,
};
```

The `diff` props is affected as {} instead of '' (empty string)! 
It's linked to `state.repository.diff`(in the `mapStateToProps` function) which is, initially, an empty string.

My question is now: Why this `diff` props isn't an empty string at the initialization of my screen?